Classes/Auth/TMOAuth.m

```diff
@@ -37,35 +37,50 @@ + (NSString *)headerForURL:(NSURL *)URL method:(NSString *)method postParameters
+    for (NSString *key in oAuthParameters)
+        [queryItems addObject:[NSURLQueryItem queryItemWithName:key value:TMURLEncode(oAuthParameters[key])]];
+
+    NSURLComponents *urlComponents = [[NSURLComponents alloc] initWithURL:URL resolvingAgainstBaseURL:true];
+    urlComponents.queryItems = queryItems;
+
+    return [urlComponents URL];
+}
+
 - (id)initWithURL:(NSURL *)URL method:(NSString *)method postParameters:(NSDictionary *)postParameters
             nonce:(NSString *)nonce consumerKey:(NSString *)consumerKey consumerSecret:(NSString *)consumerSecret
             token:(NSString *)token tokenSecret:(NSString *)tokenSecret {
+
     if (self = [super init]) {
-        NSMutableDictionary *headerParameters = [[NSMutableDictionary alloc] initWithDictionary:@{
-                                                 @"oauth_timestamp" : UNIXTimestamp([NSDate date]),
-                                                 @"oauth_nonce" : nonce,
-                                                 @"oauth_version" : @"1.0",
-                                                 @"oauth_signature_method" : @"HMAC-SHA1",
-                                                 @"oauth_consumer_key" : consumerKey,
-                                                 }];
-        
-        if (token && token.length > 0)
-            headerParameters[@"oauth_token"] = token;
-        
-        NSDictionary *queryParameters = TMQueryStringToDictionary(URL.query);
-        
-        NSString *baseURLString = [[URL absoluteString] componentsSeparatedByString:@"?"][0];
-        
-        NSString *baseString = generateBaseString(baseURLString, method, headerParameters, queryParameters, postParameters);
-        
-        _baseString = baseString;
-        
-        headerParameters[@"oauth_signature"] = sign(baseString, consumerSecret, tokenSecret);
+        NSMutableDictionary *oAuthParameters = generateOAuthParameters(self,
+                                                                       URL,
+                                                                       method,
+                                                                       postParameters,
+                                                                       nonce,
+                                                                       consumerKey,
+                                                                       consumerSecret,
+                                                                       token,
+                                                                       tokenSecret,
+                                                                       UNIXTimestamp([NSDate date]));
         
         NSMutableArray *components = [NSMutableArray array];
-        
-        for (NSString *key in headerParameters)
-            [components addObject:[NSString stringWithFormat:@"%@=\"%@\"", key, TMURLEncode(headerParameters[key])]];
+        for (NSString *key in oAuthParameters)
```

Comment:
Even for single line statements, we tend to favor adding in the curly braces for control flow statements. (I know this is existing code, so feel free to ignore this if you don't want to clean it up)